scvi/models/_base.py

```diff
@@ -0,0 +1,792 @@
+import torch
+import pickle
+import os
+import logging
+import pandas as pd
+import numpy as np
+import inspect
+
+from anndata import AnnData
+from functools import partial
+from scvi.core._distributions import NegativeBinomial, ZeroInflatedNegativeBinomial
+from scvi.models._differential import DifferentialComputation
+from scvi.models._utils import (
+    scrna_raw_counts_properties,
+    _get_var_names_from_setup_anndata,
+)
+from scvi.dataset._utils import (
+    _check_anndata_setup_equivalence,
+    _check_nonnegative_integers,
+)
+from scvi import _CONSTANTS
+from typing import Optional, Union, List, Dict
+from scvi._compat import Literal
+from scvi.core.trainers import UnsupervisedTrainer
+from abc import ABC, abstractmethod
+from scvi.dataset import get_from_registry, transfer_anndata_setup
+
+logger = logging.getLogger(__name__)
+
+
+class VAEMixin:
```

Comment:
this needs to be documented. Are we following a standard (i.e., sklearn?), what does this include, what's the point of this class?